### PR TITLE
Data jpa and another

### PR DESCRIPTION
**Pull request by ADIGrimm**

### CODE REVIEW

File: src/main/java/online/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -32,4 +35,10 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/controller/BookController.java:42]: add update implementation

File: src/main/java/online/bookstore/repository/BookRepository.java [hunk 2]
```diff
@@ -1,13 +1,9 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> findById(Long id);
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/repository/BookRepository.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/online/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -32,4 +36,16 @@ public BookDto getBookById(@PathVariable Long id) {
...
+        bookService.deleteById(id);
+    }
+
+    @ResponseStatus(HttpStatus.OK)
```
Comment (liliia-ponomarenko) [inline src/main/java/online/bookstore/controller/BookController.java]:
```suggestion
```
It is the default response status.

File: src/main/java/online/bookstore/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -37,4 +37,22 @@ public BookDto findById(Long id) {
...
+    @Override
+    public void update(Long id, CreateBookRequestDto bookRequestDto) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id " + id));
+        book.setTitle(bookRequestDto.getAuthor());
+        book.setAuthor(bookRequestDto.getAuthor());
+        book.setIsbn(bookRequestDto.getIsbn());
+        book.setPrice(bookRequestDto.getPrice());
+        book.setDescription(bookRequestDto.getDescription());
+        book.setCoverImage(bookRequestDto.getCoverImage());
```
Comment (liliia-ponomarenko) [inline src/main/java/online/bookstore/service/impl/BookServiceImpl.java]: use mapper (mapstruct) for this logic

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (liliia-ponomarenko) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: add one empty line at the end of the file ;)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! A few comments ;)

File: src/main/java/online/bookstore/controller/BookController.java [hunk 6]
```diff
@@ -32,4 +36,15 @@ public BookDto getBookById(@PathVariable Long id) {
...
     }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public void update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/controller/BookController.java]:
```suggestion
    public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
```

File: src/main/java/online/bookstore/model/Book.java [hunk 7]
```diff
@@ -30,4 +34,6 @@ public class Book {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @Column(name = "is_deleted", nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/online/bookstore/model/Book.java]:
```suggestion
    @Column(nullable = false)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by ahoienko — APPROVED** [review]
Good job!

**Pull request merged**